app/scripts/metamask-controller.js

```diff
@@ -1825,14 +1826,20 @@ export default class MetamaskController extends EventEmitter {
    * @param {Object} msgParams - The params passed to eth_sign.
    * @param {Function} cb - The callback function called with the signature.
    */
-  newUnsignedMessage(msgParams, req) {
-    const promise = this.messageManager.addUnapprovedMessageAsync(
-      msgParams,
-      req,
-    );
-    this.sendUpdate();
-    this.opts.showUserConfirmation();
-    return promise;
+  async newUnsignedMessage(msgParams, req) {
+    const data = normalizeMsgData(msgParams.data);
+    let promise;
+    // 64 hex + "0x" at the beginning
+    if (data.length === 66) {
```

Comment:
Why require messages to be 64 hex characters?

Reply:
This is the requirement of ECSign we use behind the scenes, it only works for 32 byte numbers

Reply:
I'll add more explicit documentation